karabo_data/reader.py

```diff
@@ -1009,21 +532,51 @@ def get_array(self, device, key, extra_dims=None):
         Parameters
         ----------
 
-        device: str
+        source: str
             Device name with optional output channel, e.g.
             "SA1_XTD2_XGM/DOOCS/MAIN" or "SPB_DET_AGIPD1M-1/DET/7CH0:xtdf"
         key: str
             Key of parameter within that device, e.g. "beamPosition.iyPos.value"
-            or "header.linkId". The data must be 1D in the file.
+            or "header.linkId".
         extra_dims: list of str
             Name extra dimensions in the array. The first dimension is
             automatically called 'train'. The default for extra dimensions
             is dim_0, dim_1, ...
         """
```

Comment:
Do you think it's worth to handle wild cards in the key parameters? For instance a user might want to combine all beam positions, digitizer channels or what ever of a given device in one array. Or should that be handled by the users e.g:  `data = xarray.concat([run.get_array(device, key) for key in keys], dim='youName_it')`

Reply:
This parallels the `get_series()` method to get a pandas series - they both retrieve an array of data for one key from one source.

For pandas, there's also a `get_dataframe()` method which retrieves several arrays and puts them together in a DataFrame. The equivalent for xarray is called a [DataSet](http://xarray.pydata.org/en/stable/data-structures.html#dataset). I want to have a method with wildcards to build one of these, but I've been putting it off while we try to figure out #.